lib/Echidna/ABI.hs

```diff
@@ -285,7 +294,7 @@ genAbiValueM = genWithDict (fmap toList . view constants) $ \case
 
 -- | Given a 'SolSignature', generate a random 'SolCalls' with that signature, possibly with a dictionary.
 genAbiCallM :: (MonadState x m, Has GenDict x, MonadRandom m) => SolSignature -> m SolCall
-genAbiCallM = genWithDict (fmap toList . view wholeCalls) (traverse $ traverse genAbiValueM)
+genAbiCallM abi = genWithDict (fmap toList . view wholeCalls) (traverse $ traverse genAbiValueM) abi >>= mutateAbiCall
```

Comment:
why are we doing a `mutateAbiCall` right after we `gen`?

Reply:
This is an interesting question. I couldn't find an easy way to mutate only the results of the values generated by the dictionaries, but mutating the generated values will perform the same effect. If you have a better way to do it, please let me know.